common/djangoapps/student/management/commands/bulk_unenroll.py

```diff
@@ -57,36 +57,42 @@ def unenroll_users_from_csv(self, csv_file):
         Un-enroll the given set of users provided in csv
         """
         reader = list(unicodecsv.DictReader(csv_file))
-        users_unenrolled = {}
-        for row in reader:
-            username = row['username']
-            course_key = row['course_id']
-
-            try:
-                course_id = CourseKey.from_string(row['course_id'])
-            except InvalidKeyError:
-                msg = 'Invalid course id {course_id}, skipping un-enrollement for {username}, {email}'.format(**row)
-                logger.warning(msg)
-                continue
-
-            try:
-                enrollment = CourseEnrollment.objects.get(user__username=username, course_id=course_id)
-                enrollment.update_enrollment(is_active=False, skip_refund=True)
-                if username in users_unenrolled:
-                    users_unenrolled[username].append(course_key.encode())
-                else:
-                    users_unenrolled[username] = [course_key.encode()]
-
-            except ObjectDoesNotExist:
-                msg = 'Enrollment for the user {} in course {} does not exist!'.format(username, course_key)
-                logger.info(msg)
-
-            except Exception as err:  # pylint: disable=W0703
-                msg = 'Error un-enrolling User {} from course {}: with error: {}'.format(username, course_key, err)
-                logger.error(msg, exc_info=True)
-
-        logger.info("Following users have been unenrolled successfully from the following courses: {users_unenrolled}"
-                    .format(users_unenrolled=["{}:{}".format(k, v) for k, v in users_unenrolled.items()]))
+        column_names = [name for name in reader[0].keys()]
+        if len(column_names) == 1 and 'course_id' in column_names:
```

Comment:
Instead of checking based on length, Can we pass an argument and check based on that ?

Reply:
We need to check the length in order to differentiate the between cases in which we are provided with usernames and course-id vs only course-ids(intended to un-enroll all users).

Reply:
Yeah I know. By using length code readability is't v good. It don't give any clue why we are doing this. My suggestion was to use flag e.g `unenroll_all_user`. which would make more sense while reading the code and it would look elegant as well.
```if unenroll_all_user:
        for row in reader:
                self.unenroll_all_users(course_id=row['course_id'])
       return
```

Reply:
@adeelehsan nice catch! 👍 I will modify after taking @waheedahmed review.